CHANGELOG.md

```diff
@@ -172,6 +172,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Enabled traditional/manual launching of DDP processes through `LOCAL_RANK` and `NODE_RANK` environment variable assignments ([#7480](https://github.com/PyTorchLightning/pytorch-lightning/pull/7480))
 
 
+- Added support for `devices` flag to Trainer ([#8440](https://github.com/PyTorchLightning/pytorch-lightning/pull/8440))
+
+
```

Comment:
this means you will deprecate gpus, ipus, tpu_cores, num_processes?

Reply:
Nooo! I don't think we are deprecating gpus, ipus, tpu_cores. `num_processes`, yes.
Related [comment](https://github.com/PyTorchLightning/pytorch-lightning/issues/6090#issuecomment-852072254)

Reply:
Yes ok but I had to ask because only a couple people have discussed in #6090 and I just wanted to make sure the team is ok with adding this Trainer argument. 

`devices` will be a Trainer argument that is not orthogonal to the others.

Reply:
I'm confused @kaushikb11. In a comment above you said

> We are planning to deprecate num_processes soon, and devices would be the go-to flag and would have 1 default value.

But here you said

> Nooo! I don't think we are deprecating gpus, ipus, tpu_cores. num_processes, yes.

So are we or are we not?

IMO we should deprecate it, but potentially set a later removal version than usual.

Reply:
@carmocca Oops, I mentioned `num_processes, yes.` in my comment above.
Yes, we will be deprecating `num_processes`.

Reply:
> IMO we should deprecate it, but potentially set a later removal version than usual.
+1 but for deprecating also `gpus`, `tpu_cores` and `ipus`. Just keep it for 2-3 major releases, but we try to unify the api here. If you only deprecate `num_processes` and nothing else, this won't be much of a gain, just a rewording of the flag.

Reply:
+1 for unification, and also fine for 3 feature releases

Reply:
Yup, will be doing follow-up PRs.